diff --git a/client/firewall/uspfilter/localip.go b/client/firewall/uspfilter/localip.go
index ffc807f4..0167564b 100644
--- a/client/firewall/uspfilter/localip.go
+++ b/client/firewall/uspfilter/localip.go
@@ -7,6 +7,7 @@ import (
 	"sync"
 
 	log "github.com/sirupsen/logrus"
+	"github.com/wlynxg/anet"
 
 	"github.com/netbirdio/netbird/client/firewall/uspfilter/common"
 )
@@ -87,7 +88,7 @@ func (m *localIPManager) processIP(ip netip.Addr, bitmap *[256]*ipv4LowBitmap, i
 }
 
 func (m *localIPManager) processInterface(iface net.Interface, bitmap *[256]*ipv4LowBitmap, ipv4Set map[netip.Addr]struct{}, ipv4Addresses *[]netip.Addr) {
-	addrs, err := iface.Addrs()
+	addrs, err := anet.InterfaceAddrsByInterface(&iface)
 	if err != nil {
 		log.Debugf("get addresses for interface %s failed: %v", iface.Name, err)
 		return
@@ -140,7 +141,7 @@ func (m *localIPManager) UpdateLocalIPs(iface common.IFaceMapper) (err error) {
 		}
 	}
 
-	interfaces, err := net.Interfaces()
+	interfaces, err := anet.Interfaces()
 	if err != nil {
 		log.Warnf("failed to get interfaces: %v", err)
 	} else {
diff --git a/client/internal/routemanager/systemops/systemops_generic.go b/client/internal/routemanager/systemops/systemops_generic.go
index ec219c7f..bc52562c 100644
--- a/client/internal/routemanager/systemops/systemops_generic.go
+++ b/client/internal/routemanager/systemops/systemops_generic.go
@@ -398,6 +398,7 @@ func ipToAddr(ip net.IP, intf *net.Interface) (netip.Addr, error) {
 
 // IsAddrRouted checks if the candidate address would route to the vpn, in which case it returns true and the matched prefix.
 func IsAddrRouted(addr netip.Addr, vpnRoutes []netip.Prefix) (bool, netip.Prefix) {
+	return false, netip.Prefix{}
 	localRoutes, err := hasSeparateRouting()
 	if err != nil {
 		if !errors.Is(err, ErrRoutingIsSeparate) {
diff --git a/client/internal/stdnet/discover_pion.go b/client/internal/stdnet/discover_pion.go
index 83b4a1d1..5e164482 100644
--- a/client/internal/stdnet/discover_pion.go
+++ b/client/internal/stdnet/discover_pion.go
@@ -1,9 +1,8 @@
 package stdnet
 
 import (
-	"net"
-
 	"github.com/pion/transport/v3"
+	"github.com/wlynxg/anet"
 )
 
 type pionDiscover struct {
@@ -12,7 +11,7 @@ type pionDiscover struct {
 func (d pionDiscover) iFaces() ([]*transport.Interface, error) {
 	ifs := []*transport.Interface{}
 
-	oifs, err := net.Interfaces()
+	oifs, err := anet.Interfaces()
 	if err != nil {
 		return nil, err
 	}
@@ -20,7 +19,7 @@ func (d pionDiscover) iFaces() ([]*transport.Interface, error) {
 	for _, oif := range oifs {
 		ifc := transport.NewInterface(oif)
 
-		addrs, err := oif.Addrs()
+		addrs, err := anet.InterfaceAddrsByInterface(&oif)
 		if err != nil {
 			return nil, err
 		}
diff --git a/client/system/info.go b/client/system/info.go
index 01176e76..8cc1a5b2 100644
--- a/client/system/info.go
+++ b/client/system/info.go
@@ -7,6 +7,7 @@ import (
 	"strings"
 
 	log "github.com/sirupsen/logrus"
+	"github.com/wlynxg/anet"
 	"google.golang.org/grpc/metadata"
 
 	"github.com/netbirdio/netbird/shared/management/proto"
@@ -146,7 +147,7 @@ func extractDeviceName(ctx context.Context, defaultName string) string {
 }
 
 func networkAddresses() ([]NetworkAddress, error) {
-	interfaces, err := net.Interfaces()
+	interfaces, err := anet.Interfaces()
 	if err != nil {
 		return nil, err
 	}
@@ -156,7 +157,7 @@ func networkAddresses() ([]NetworkAddress, error) {
 		if iface.HardwareAddr.String() == "" {
 			continue
 		}
-		addrs, err := iface.Addrs()
+		addrs, err := anet.InterfaceAddrsByInterface(&iface)
 		if err != nil {
 			continue
 		}
